### PR TITLE
Clojure 008

### PR DESCRIPTION
**Pull request by yvendruscolo**
### How the solution works

defining big-numbers as a string so I can cut it up in chars, put them all in a vector, then read them into integers again.

then use 'partition to build up a lazy seq of a window of size 13 and step size 1.
this then is transduced so that we can find the product of each seq, but only keeping the largest before the next step.

### Performance

[Try it online!](https://tio.run/##bZTLbls3EIb3foqDrKQCAeZKzjxLkIViq4YKWzJkuejbOx8dt120BKRzITnz33juny5/vF2P7@@7h@Pv5@318vTncfv2/W7bdk/H2/btx@nx6/nt@cfxun2ZrmPqHOnqLsOitc1StceMsJEzZphn@pqJ5jK6PE1cbc5IGW7DvHOWa4kyG1xqdMyq1AqvrKE5ZFarGkWiM7JTdPrsLHfttEo3qZKkuVqGUInmmpXmNVxSeBjt1tI0ZYsB0SVAN8eoHiOKqhFp6wczGmjlCHGh2Kqr06jmqj7MrAtC5l4sKPV03lJKw8pKRjCKqgvAhLt4Vsu0HmBADzhFyJzQB5uXSi64LJw9hJ0TihY26VvlDYaeLfwv6G3aS6@ei3dJ@6R3TpuOFUIx4KChmgkOBDWkZy5tUCfRFwUgb1lBf@jRL8EENwnaIXsE7xArQilsio@U8NGzFCyzAo2QGkdSQQS4YR8ia69JnizQgl3cqM8EoNCDXLQYSkF1CVID0TXSU0WITJmO4RMcKIS/KC0CTVwLX6lBp4DZzEKnAUsdS6ipMbvBG9aYgC3DYyRhxB8SqMtWJmJlan6YQxp5HYl2hIsocwtksqCG6phJ9AbtAwpLczMiv5QylNE1lvaS9NQJtA4FHg6GFL4SaCHDhI4mLApC0rjykalaaPGYxCBIY3OBSIgG/LFrQSZfM4N1IrDHDrIKtnUiOHmCmGiw@C9YHSvorEdHW1bAIDlu7CVYmIjcOZUiNpouVbCRFVhqF84LqrEiORgmRISzgt5fOP2/xsPp8XR73Xan8@3CV2H7Z@LvsXs@vGy7@8vzy3Y9Hh6@vt6up/PjxmW//@/qfz8m@@@fk7vb9XB@fXi7P37Wejlcb6fDE@U@Xv72f3WeD39t8rn0drqcN/VNP8HuGXd3u5fredt9fM32@/f3nw "Clojure – Try It Online")

```
Real time: 3.108 s
User time: 5.636 s
Sys. time: 0.193 s
CPU share: 187.58 %
```

### CODE REVIEW

**Comment by caian-gums** [comment]
Here too, if you can place a link to Try it Online it will be nice! [Try it online!](https://tio.run/##bZTdcuNEEIXv/RRDbrAoAtM93fPDXvMUqVw4tpJ1cGQjy7B5@vCNE2CpRVW2pJlW9@lzzvT2cHy@zOPb26fw53H@7Rz2U3g4Lp/D9n0jbKbd38/n7bw/LavVeplfwyqE9Tz@ftkT8/3d9vB8/mkeN7txDr/M4yO3u/56e17m/fR0fz/0@O1m2X4Ov37Zjqdlf5zCGKb94brzuJ82h8NrWJ8IXw5TuNkdp6VDml9/DPvlHJ4v5yU8HmfAPP/cy4Xt8eW0Wb67GYZhtfoUrmu743jmu81lOd7@g@4rIIDfjY/hYf90O11eHoB5U5LkIiV7kpRiVmvSVF2k5WKm2YsV0@Se@o41brnV5BqTaCnmMSfNmpqXmqRGYde41dys1OpSLVWvWTzHUpuIksSamzePUlJpXlOS5lo9aaw1OsVF3SKZKC5eXVPNKXrkJbekLTaK8okCMUUDXcm5tpytktXMtf/ojAJSPVtMkWQ9rxQlWxJJWVVbpSFNqRJQJXlilVRiWrXGbFyVrB1AofeYvLZYtGUwwAc9mcVSaB9sqUr0DpfA0nLky0KLalqoW2tqYGilRf479KbSOl@t9L5rbKlQ24uWhBSRZMCBQ1GNKGDkiK145wZ2HH5hgObVq1Gf9qjnYKK3aJSDdjPWIMtMSKyCjqRIuZUqYCnV4AiqUcQFRIDLeiVZWt/kTQ0u@IoHScUBGKmBL1pUmKLVTkjNkC7mySVGLFNVck4FHDCEvjAdI22imqXuGngyOite4SnTpeROVBErrYHXtCECsuRk2TEj@uBA6bKyYd1T5SoObmTZHO4wF1bmEch4QRTWERPrZcobLXTOVbF8Z0phRvrVuY9OTSlAaybAQ0GLFV0xdMTDmI4iBBkmaahy9VTtaNEYx0BIQ@YKoog16B@5OmT8VdyIi5HukQOvgq2fCE5ehEw46P13WM260YmHR@1S0IFz3PgWYyEidHsRkmhuVKmVbmI3LLkrykdYI8I5GBqxCGcFvm@G98M/hfPx8McY7u774DmMS7jb7Z/6iFkzeo6s99n232v9sjkxvhg44athErgNw7fR/06X4f5jk5G5mc67y3b8yHXazMt@cyDddfGH/8vzsvkS4kfodV5KChLewQ7Xqce4nML62s8wvL39BQ "Clojure – Try It Online")

And if you can clear your code in order to just print the result of the problem? 😄

File: src/008/p008.cljc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+; works in both clojure and clojurescript
+
+(try 
+  (require '[cljs.reader :refer [read-string]])
+  (catch Exception e nil)
+  (finally (println "dont worry, its just for clj/cljs compat!")))
```
Comment (caian-gums) [inline src/008/p008.cljc]: Is this print necessary?

**Comment by yvendruscolo** [comment]
that print is not actually necessary, but if we try to run it from cljs, it will "complain" about the Exception in the catch expression.

I could remove the print, but it will still print a warning when run in cljs.
Another thing that can be done is splitting it into 2 files, one for clj and other for cljs.
Should I?

**Comment by FrankKair** [comment]
I think we should make each individual language run on its own platform.
So for me, Clojure would run solely as Clojure, not ClojureScript.

One of the ideias of the repo is provide a good base for comparison between languages, and I personally think that adding multiple targets / environments to a file puts some cognitive load on someone just trying to read Clojure to learn it.

What do you think?

**Comment by yvendruscolo** [comment]
well, I could keep them as one while they remain the same, and split them up whenever something like this comes up.

**Comment by FrankKair** [comment]
My opinion is that we should focus only on Clojure. I'd like to know what @caian-gums and @fredericojordan think about it.

**Comment by FrankKair** [comment]
Also, don't forget to rebase to sync with the master branch:

```
$ git checkout master
$ git pull
$ git checkout 008cljc
$ git rebase master
$ git push -f
```

**Comment by FrankKair** [comment]
Nice! ✅

**Pull request merged**